File: backend/src/main/java/it/unical/inf/ea/backend/data/services/interfaces/CartService.java
package it.unical.inf.ea.backend.data.services.interfaces;

import it.unical.inf.ea.backend.dto.CartDTO;
import it.unical.inf.ea.backend.dto.creation.CartItemCreateDTO;

import java.util.UUID;

public interface CartService {
    CartDTO getCartByUserId(UUID userId);
    CartDTO getCartForLoggedUser();
    CartDTO addItemToCart(CartItemCreateDTO cartItem);
    CartDTO removeItemFromCart(UUID cartItemId);
    CartDTO clearCart();
}

File: Progetto-2023-2024-Vecchio-Larocca-Lentini-Mahrous/frontend/app/src/main/java/com/android/frontend/service/CartService.kt
package com.android.frontend.service

import com.android.frontend.dto.creation.CartCreateDTO
import com.android.frontend.dto.CartDTO
import com.android.frontend.dto.creation.CartItemCreateDTO
import retrofit2.Call
import retrofit2.http.*
import java.util.UUID

interface CartService {

    @GET("/api/v1/cart/getCartByUserId/{userId}")
    fun getCartByUserId(
        @Path("userId") userId: UUID
    ): Call<CartDTO>

    @GET("/api/v1/cart/getCartForLoggedUser")
    fun getCartForLoggedUser(
        @Header("Authorization") token: String
    ): Call<CartDTO>


    @PUT("/api/v1/cart/editItem")
    fun editItemInCart(
        @Header("Authorization") token: String,
        @Query("cartItemId") cartItemId: UUID,
        @Query("quantity") quantity: Int
    ): Call<CartDTO>

    @POST("/api/v1/cart/addItem")
    fun addItemToCart(
        @Header("Authorization") token: String,
        @Body cartItemCreateDTO: CartItemCreateDTO
    ): Call<CartDTO>

    @DELETE("/api/v1/cart/removeItem/{cartItemId}")
    fun removeItemFromCart(
        @Header("Authorization") token: String,
        @Path("cartItemId") cartItemId: UUID
    ): Call<CartDTO>

    @DELETE("/api/v1/cart/clearCart")
    fun clearCart(
        @Header("Authorization") token: String
    ): Call<CartDTO>
}
